cmd/registry/controller/parser.go

```diff
@@ -157,39 +158,78 @@ func resourceNameFromParent(
 	return resource, nil
 }
 
-func ValidateResourceEntry(resource *rpc.GeneratedResource) error {
-	var group string
+func ValidateManifest(ctx context.Context, parent string, manifest *rpc.Manifest) []error {
+	totalErrors := make([]error, 0)
+	for _, resource := range manifest.GeneratedResources {
+		if errs := validateGeneratedResourceEntry(parent, resource); len(errs) != 0 {
+			for _, err := range errs {
+				totalErrors = append(totalErrors, fmt.Errorf("invalid entry: %v, %s", resource, err))
+			}
+		}
+	}
+	return totalErrors
+}
+
+func validateGeneratedResourceEntry(parent string, generatedResource *rpc.GeneratedResource) []error {
+	errs := make([]error, 0)
+	parsedTargetResource, err := parseResourcePattern(
+		fmt.Sprintf("%s/%s", parent, generatedResource.Pattern))
+
+	// Check that the target resource pattern should be a valid pattern.
+	// Return for errors in target resource pattern since we caan't verify action and dependencies based off an incorrect pattern.
+	if err != nil {
+		errs = append(errs, fmt.Errorf("invalid pattern for generatedResource %v, %s", generatedResource.Pattern, err))
+		return errs
```

Comment:
Nit: Consider declaring the single element slice as it's returned to make it clearer that this is the only error that will be returned. `errs` can be declared before the loop where multiple errors might actually be returned.

```
return []error{
    fmt.Errorf("invalid pattern for generatedResource %v, %s", generatedResource.Pattern, err),
}
```

Same goes for the suffix check below.

Reply:
Done.